src/client.ts

```diff
@@ -393,15 +414,15 @@ export class Client extends EventEmitter {
       }
 
       this.activeStanzaInstance = stanzaInstance;
+      stanzaInstance.pinger = new Ping(this, stanzaInstance);
       this.emit('connected');
     } catch (err) {
       if (stanzaInstance) {
         this.logger.error('Error occurred in connection attempt, but after websocket connected. Cleaning up connection so backoff is respected', { stanzaInstanceId: stanzaInstance.id, channelId: stanzaInstance.channelId });
+        this.removeStanzaBoundEventHandlers();
 
-        // unproxy stanza events so we don't try and reconnect
-        (stanzaInstance as any).emit = stanzaInstance.originalEmitter;
-        stanzaInstance.pinger!.stop();
-        stanzaInstance.disconnect();
+        stanzaInstance.pinger?.stop();
+        await (stanzaInstance as unknown as StanzaClient).disconnect();
```

Comment:
I'm assuming the reason there's two types here is because we could have no `stanzaInstance` and `StanzaClient` doesn't support that?

Reply:
No the reason here is *crraazzzzyyyy*. It's stanza's fault. bad typing. From stanza's index file:
```
export function createClient(opts: AgentConfig): Agent {
    const client = new Client(opts);
    client.use(Plugins);

    return (client as unknown) as Agent;
}
```

The `Agent` type has a `disconnect()` fn, but it returns void. the `Client` class has a `disconnect()` fn, but it returns a promise. I want to await the promise.

Reply:
Ew.